keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensor.java

```diff
@@ -1045,9 +1045,14 @@ public BooleanTensor greaterThanOrEqual(DoubleTensor value) {
 
     @Override
     public BooleanTensor elementwiseEquals(Tensor that) {
-
-        if (that instanceof Nd4jDoubleTensor) {
-            INDArray eq = tensor.eq(unsafeGetNd4J((Nd4jDoubleTensor) that));
+        INDArray eq;
```

Comment:
I'm not a fan of uninitialised variables. I think this is a scenario where the `? :` syntax is preferable:

        if (that instanceof DoubleTensor) {
            INDArray eq = that.isScalar() ?
                tensor.eq(((DoubleTensor) that).scalar()) : 
                tensor.eq(unsafeGetNd4J((Nd4jDoubleTensor) that));
            return fromMask(eq, getShape());

Reply:
likewise in `Nd4jIntegerTensor.java`